source/configuration_manual/event_filter.rst

```diff
@@ -0,0 +1,119 @@
+.. _event_filter:
+
+===============
+Event Filtering
+===============
+
+Dovecot's event support includes the ability to narrow down which events are
+processed by filtering them based on the administrator-supplied predicate.
+
+Individual events can be identified either by their name or source code
+location.  The source location of course can change between Dovecot
+versions, so it should be avoided.
+
+Regardless of the syntax used, matching is performed the same way:
+
+* Event names are compared using a case-sensitive wildcard match.  The
+  wildcards supported are ``?`` and ``*``.
+* Event location is compared in two parts: the file name is compared
+  case-sensitively, and the line number is compared as an integer.  For a
+  match to occur, the filename match *and* the line number must either match
+  or be unspecified.
+* Event categories are compared using a "has a" relationship.  A category in
+  the filter must be present in the event for a match to occur.  Any other
+  categories on the event do not influence the match.
```

Comment:
Should it say something about event hierarchies?

Reply:
I'm not sure.  The event and category parents are a code-level/implementation detail.  The sysadmin doesn't see the hierarchy.  All that is visible is the flattened event that event exporting spits out.